packages/datadog-plugin-aws-sdk/src/services/sqs.js

```diff
@@ -30,9 +33,11 @@ class Sqs extends BaseAwsSdkPlugin {
             { 'span.kind': 'server' }
           )
         }
-        const span = plugin.tracer.startSpan('aws.response', options)
+        span = plugin.tracer.startSpan('aws.response', options)
         this.enter(span, store)
       }
+      // extract DSM context after as we might not have a parent-child but maybe have a DSM context
+      this.responseExtractDSMContext(request.operation, request.params, response, span ?? null)
```

Comment:
This ends up parsing the JSON twice.

Reply:
The caveat here is that for normal trace context extraction, we only perform it if we receive 1 message, not for batches. However, for DSM, we should be setting a checkpoint for each message received. I have updated the code logic to be:
- If 1 message is received, we parse the message attributes for a trace context and return both the trace context and parsed attributes. When looking for a DSM context, the parsed attributes already exist and we do not parse again.
- In the case of a batch receive, we do not parse the message attributes looking for a trace context since we have more than 1 message. In this case, we parse the JSON within the DSM context extraction function since no parsed attributes were passed into the function.